Orchestrator/Infrastructure/Auth/AssetAccessValidator.cs

```diff
@@ -0,0 +1,106 @@
+    /// </summary>
+    public class AssetAccessValidator
+    {
+        private readonly ISessionAuthService sessionAuthService;
+        private readonly AccessChecker accessChecker;
+        private readonly AuthCookieManager cookieManager;
+        private readonly IHttpContextAccessor httpContextAccessor;
+
+        public AssetAccessValidator(
+            ISessionAuthService sessionAuthService,
+            AccessChecker accessChecker,
+            AuthCookieManager cookieManager,
+            IHttpContextAccessor httpContextAccessor)
+        {
+            this.sessionAuthService = sessionAuthService;
+            this.accessChecker = accessChecker;
+            this.cookieManager = cookieManager;
+            this.httpContextAccessor = httpContextAccessor;
+        }
+
+        /// <summary>
+        /// Validate whether Bearer token associated with provided request has access to the specified roles for
+        /// customer.
+        /// This will add SET-COOKIE header to response if the  asset is restricted and the current request
+        /// has access. The Response StatusCode is NOT altered here - use the returned enum to do this downstream.
+        /// </summary>
+        /// <param name="customer">Current customer</param>
+        /// <param name="roles">Roles associated with Asset</param>
+        /// <returns><see cref="AssetAccessResult"/> enum representing result of validation</returns>
+        public async Task<AssetAccessResult> TryValidateBearerToken(int customer, IEnumerable<string> roles)
+        {
+            var assetRoles = roles.ToList();
+            if (assetRoles.IsNullOrEmpty()) return AssetAccessResult.Open;
+
+            var httpContext = httpContextAccessor.HttpContext.ThrowIfNull(nameof(httpContextAccessor.HttpContext))!;
+            
+            var bearerToken = GetBearerToken(httpContext.Request);
+            if (string.IsNullOrEmpty(bearerToken))
+            {
+                // No bearer token, 401 but call underlying (e.g. to render info.json)
+                return AssetAccessResult.Unauthorized;
+            }
+            
+            // Get the authToken from bearerToken
+            var authToken =
+                await sessionAuthService.GetAuthTokenForBearerId(customer, bearerToken);
+
+            if (authToken?.SessionUser == null)
+            {
+                // Bearer token not found, or expired, 401 but call underlying (e.g. to render info.json)
+                return AssetAccessResult.Unauthorized;
+            }
+            
+            // Validate current user has access for roles for requested asset
+            var canAccess = await accessChecker.CanSessionUserAccessRoles(authToken.SessionUser, customer, assetRoles);
+
+            if (!canAccess)
+            {
+                return AssetAccessResult.Unauthorized;
+            }
+            
+            cookieManager.SetCookieInResponse(authToken);

```

Comment:
DLCS validates access for 2 types of request:
 - info.json requests made by script that won't include any cookies the user might have but might include a bearer token in the authorisation header - where the client acquired this token via postMessage when it opened the token service in an iframe. Direct HEAD requests are also made this way by script clients because there's no info.json for AV
 - image/av requests made by anything else - e.g., img tag, openseadragon - which will include any cookies the user has acquired.

Image requests won't have a bearer token in them (how would an img tag set that?). At least in auth 1.0 they won't.
So I think this line can just be removed. In fact it's less secure; a client that only has a bearer token (maybe acquired with bad intent) could cause the DLCS to set a cookie for the image.